packages/desktop-client/src/components/settings/UI.js

```diff
@@ -53,8 +55,21 @@ export function ButtonSetting({ button, children }) {
 
 export function AdvancedToggle({ children }) {
   let [expanded, setExpanded] = useState(false);
+  let location = useLocation();
+  useEffect(() => {
+    if (location.hash === '#advanced') {
+      setExpanded(true);
+    }
+  }, [location.hash]);
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏moving logic to `useEffect` will cause an unnecessary render. Basically the first render will always be with `expanded = false` and then the second will set it to `expanded = true` (if `#advanced` is set).

We should be able to cut this down to 1x by doing..

```suggestion
  let location = useLocation();
  let [expanded, setExpanded] = useState(location.hash === '#advanced');
```

Reply:
Makes sense! And I guess there isn’t much risk of changing the `location.hash` state later on, so it’s safe to not check for updates.